composer/models/ssd/ssd.py

```diff
@@ -0,0 +1,173 @@
+import contextlib
+import logging
+import os
+import random
+import time
+from argparse import ArgumentParser
+from typing import Any, Optional, Sequence, Tuple
+
+import numpy as np
+import torch
+from PIL import Image
+from pycocotools.cocoeval import COCOeval
+from torch.utils.data import DataLoader
+from torchmetrics import Metric
+from torchmetrics.detection.map import MeanAveragePrecision as MAP
+
+from composer.core.types import BatchPair, Metrics, Tensor, Tensors
+from composer.datasets.coco import COCO, COCODetection
+from composer.models.base import ComposerModel
+from composer.models.ssd.base_model import Loss
+from composer.models.ssd.ssd300 import SSD300
+from composer.models.ssd.ssd_hparams import SSDHparams
+from composer.models.ssd.utils import DefaultBoxes, Encoder, SSDTransformer
+
+
+class SSD(ComposerModel):
+
+    def __init__(self, hparams: SSDHparams) -> None:
+        super().__init__()
+
+        self.hparams = hparams
+        ln = COCODetection.labelnum
+        self.module = SSD300(80, model_path="/mnt/r1z1/laura/composer/resnet34-333f7ec4.pth")
+        ##todo(laura): fix weights path
+        dboxes = dboxes300_coco()
+
+        self.loss_func = Loss(dboxes)
+        self.MAP = MAP()
+
+    def loss(self, outputs: Any, batch: BatchPair) -> Tensors:
+
+        (img, img_id, img_size, bbox, label) = batch
+        trans_bbox = bbox.transpose(1, 2).contiguous()
+
+        ploc, plabel = outputs
+        gloc, glabel = transbbox, label
+        '''
+        Variable(trans_bbox, requires_grad=False), \
+                        Variable(label, requires_grad=False)
+        '''
+
+        loss = self.loss_func(ploc, plabel, gloc, glabel)
+        return loss
+
+    def metrics(self, train: bool = False) -> Metrics:
+
+        return self.MAP
+
+    def forward(self, batch: BatchPair) -> Tensor:
+        (img, img_id, img_size, bbox, label) = batch
+
+        ploc, plabel = self.module(img)
+
+        return ploc, plabel
+
+    def validate(self, batch: BatchPair) -> Tuple[Any, Any]:
+        dboxes = dboxes300_coco()
+        input_size = 300
+        val_trans = SSDTransformer(dboxes, (input_size, input_size), val=True)
+        data = "/localdisk/coco"
+
+        val_annotate = os.path.join(data, "annotations/instances_val2017.json")
+        val_coco_root = os.path.join(data, "val2017")
+        train_annotate = os.path.join(data, "annotations/instances_train2017.json")
+        train_coco_root = os.path.join(data, "train2017")
+        val_coco = COCODetection(val_coco_root, val_annotate, val_trans)
+
+        inv_map = {v: k for k, v in val_coco.label_map.items()}
```

Comment:
I feel like there has to be a better way to get the proper labels than to create a `Dataset` object within the `validate()` function. I think the `__getitem__` function could return whatever value is being extracted from this conversion. We can discuss more in person